### DISCUSSION TITLE
Esp32 S3 USB Hub

### DISCUSSION THREAD

**Discussion by maaamcube** [hathach/tinyusb#2467] [Q&A]
I am working on the ESP32-S3 boards to enable connection of my STM32 boards in device mode (3 boards) interconnected with hubs via USB (vendor specific) to establish a WiFi communication with PC software.
I tested with an example from the ESP IDF host library to connect these boards, and only the first hub is recognized, but I see on the GitHub of TinyUSB that support for multiple devices through a hub is available. So, my question is: where can I find the code that does this, and how can I integrate it into the example of the ESP IDF?
Thanks.

**Comment by leeebo** [comment]
`I tested with an example from the ESP IDF host library to connect these boards, and only the first hub is recognized,`

@maaamcube  Regrettably, the ESP IDF host library does not currently support external Hubs. If you are interested in staying informed about the progress of this development, please subscribe to this esp-idf issue https://github.com/espressif/esp-idf/issues/12554.

**Reply by maaamcube** [reply]
@leeebo when it will be possible, can i easily add that to the esp iot bridge framework ?
